data-solutions/data-platform-foundations/02-resources/README.md

```diff
@@ -38,15 +38,15 @@ project_ids = {
 }
 ```
 
-- Get a key for the service account created in the environment stage:
-  - Go into services project
-  - Go into IAM page
-  - Go into the service account section
-  - Creaet a new key for the service account created in previeous step (**service_account**)
-  - Download the json key into the current folder
-- make sure you have the right authentication setup: `export GOOGLE_APPLICATION_CREDENTIALS=PATH_TO_SERVICE_ACCOUT_KEY.json`
-- run `terraform init` and `terraform apply`
 
+- Set up the gcloud environment to impersonate the **service_account**: `gcloud config set auth/impersonate_service_account **service_account**`
+
+- Generate an access token and launch terraform:
+```bash
+export GOOGLE_OAUTH_ACCESS_TOKEN=$(gcloud auth print-access-token)
```

Comment:
the GCP provider supports this natively now through the [impersonate_service_account](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#impersonate_service_account) option.